### PR TITLE
GPU power utilization added to monitoring namespace.

### PR DESCRIPTION
**Pull request by harishankar-gopalan**
Hoping updation of CHANGELOG.md and docs update will be done from the Neptune team side.
Fixes the issue in https://github.com/neptune-ai/neptune-client/issues/1853

### CODE REVIEW

**Comment by SiddhantSadangi** [comment]
Thanks for this PR @harishankar-gopalan 💟 

We've handled the internals (changelog, tests, etc) and made a few changes to the implementation (replaced `nvmlDeviceGetPowerManagementLimit` with `nvmlDeviceGetEnforcedPowerLimitConstraints` for broader support and relevancy).

We'll merge and release sometime next week.

Meanwhile, would it be possible for you to check if all changes work as expected at your end?

**Comment by harishankar-gopalan** [comment]
Sure @SiddhantSadangi I would do that in the coming week and get back.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ### Changes
 - Added docstring for the `pop()` function ([#1781](https://github.com/neptune-ai/neptune-client/pull/1781))
-
+- Added GPU power consumption monitoring ([#1854](https://github.com/neptune-ai/neptune-client/pull/1854))
```
Comment (Raalsky) [inline CHANGELOG.md:5]: Let's move the version to `1.11.0` as it's a feature & place it under "Features" section (needs to be created).

File: src/neptune/common/hardware/constants.py [hunk 2]
```diff
@@ -13,7 +13,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-__all__ = ["BYTES_IN_ONE_MB", "BYTES_IN_ONE_GB"]
+__all__ = ["BYTES_IN_ONE_MB", "BYTES_IN_ONE_GB", "WATTS_TO_MILLIWATTS"]
```
Comment (Raalsky) [inline src/neptune/common/hardware/constants.py:16]: Could you rename it to `MILIWATTS_IN_WATT` instead? For readability and to keep the pattern 😉

File: src/neptune/common/hardware/gpu/gpu_monitor.py [hunk 3]
```diff
@@ -43,6 +45,24 @@ def get_card_usage_percent(self, card_index):
     def get_card_used_memory_in_bytes(self, card_index):
         return self.__nvml_get_or_else(lambda: nvmlDeviceGetMemoryInfo(nvmlDeviceGetHandleByIndex(card_index)).used)
 
+    def get_card_power_usage(self, card_index):
+        return self.__nvml_get_or_else(
+            lambda: int(nvmlDeviceGetPowerUsage(nvmlDeviceGetHandleByIndex(card_index)) // WATTS_TO_MILLIWATTS)
```
Comment (Raalsky) [inline src/neptune/common/hardware/gpu/gpu_monitor.py:50]: Could you please move the units changing to the Gauge-level (made it a part of `GpuPowerGauge`) for consistency.

File: src/neptune/common/hardware/gpu/gpu_monitor.py [hunk 4]
```diff
@@ -43,6 +45,22 @@ def get_card_usage_percent(self, card_index):
     def get_card_used_memory_in_bytes(self, card_index):
         return self.__nvml_get_or_else(lambda: nvmlDeviceGetMemoryInfo(nvmlDeviceGetHandleByIndex(card_index)).used)
 
+    def get_card_power_usage(self, card_index):
+        return self.__nvml_get_or_else(lambda: nvmlDeviceGetPowerUsage(nvmlDeviceGetHandleByIndex(card_index)))
+
+    def get_card_max_power_rating(self):
```
Comment (Raalsky) [inline src/neptune/common/hardware/gpu/gpu_monitor.py:51]: Unused?

Comment (SiddhantSadangi) [reply src/neptune/common/hardware/gpu/gpu_monitor.py:51]: Nope. Used in [`SystemResourceInfoFactory.__create_whole_system_resource_info()`](https://github.com/neptune-ai/neptune-client/blob/6bb4a9dc2be8a41a03e4e0e3c84db4370be9278c/src/neptune/common/hardware/resources/system_resource_info_factory.py#L46) and [`SystemResourceInfoFactory.__create_whole_system_resource_info()`](https://github.com/neptune-ai/neptune-client/blob/6bb4a9dc2be8a41a03e4e0e3c84db4370be9278c/src/neptune/common/hardware/resources/system_resource_info_factory.py#L57)

Comment (Raalsky) [reply src/neptune/common/hardware/gpu/gpu_monitor.py:51]: Ok, thanks

**Review by Raalsky — APPROVED** [review]

**Comment by SiddhantSadangi** [comment]
@harishankar-gopalan - this has been included in the [1.11.0 release](https://github.com/neptune-ai/neptune-client/releases/tag/1.11.0) 🎉

**Pull request merged**